src/functions/updateUploadStatus/framework/handler.ts

```diff
@@ -38,15 +39,27 @@ export async function handler(event: APIGatewayProxyEvent, fnCtx: Context): Prom
     await updateUpload(appRef, body);
   } catch (err) {
     if (err instanceof InconsistentUpdateError) {
-      warn('InconsistentUpdateError', err);
+      warn('InconsistentUpdateError - ', ...enrichError(err, appRef, body));
       return createResponse(
         { message: `Failed to update a single record for application reference ${appRef}` },
         HttpStatus.NOT_FOUND,
       );
     }
-    error(err);
+    error('Error while updating upload status - ' , ...enrichError(err, appRef, body));
     return createResponse(
       { message: `Error updating the status in UUS of Reference Number: ${appRef}` }, HttpStatus.INTERNAL_SERVER_ERROR);
   }
   return createResponse({}, HttpStatus.CREATED);
 }
+
+function enrichError(err: any, applicationReference: number, body: SubmissionOutcome) {
+  return {
+    ...err,
+    applicationReference,
+    uploadStatus: body.state,
```

Comment:
those `body.XXX` properties - will they be always present regardless of how the execution will flow? (happy paths vs edge cases) - do we need some defensive programming here?

Reply:
They should do every previous line has an error condition and returns a response so the only way we should hit this catch is if there is a SQL issue.

I have added some defenses here just in case.